src/Symfony/Bridge/Twig/Mime/BodyRenderer.php

```diff
@@ -45,15 +45,13 @@ public function render(Message $message): void
             return;
         }
 
-        $messageContext = $message->getContext();
-
-        $previousRenderingKey = $messageContext[__CLASS__] ?? null;
-        unset($messageContext[__CLASS__]);
-        $currentRenderingKey = $this->getFingerPrint($message);
-        if ($previousRenderingKey === $currentRenderingKey) {
+        if (null === $message->getTextTemplate() && null === $message->getHtmlTemplate()) {
```

Comment:
I would recommend using the method TemplatedEmail::isRendered() the same way it is used [here](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/Mailer/Transport/AbstractTransport.php#L83).

Keeping like this prevents subclass of TemplatedEmail with computed html template to not be rendered twice, such as [NotificationEmail](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Bridge/Twig/Mime/NotificationEmail.php#L202).

Reply:
Please create a PR or at least a new issue, thanks.

Reply:
Thank your for the follow-up @OskarStark , here is the PR : https://github.com/symfony/symfony/pull/59596